test/types.ts

```diff
@@ -0,0 +1,31 @@
+import { AbortController as AbortControllerShim, AbortSignal as AbortSignalShim } from "..";
+
+function signalCallback(as: AbortSignal) { }
+function signalShimCallback(as: AbortSignalShim) { }
+function controllerCallback(ac: AbortController) { }
+function controllerShimCallback(ac: AbortControllerShim) { }
+function abortCallback(this: AbortSignal, ev: Event) { }
+
+const controller = new AbortController();
+const controllerShim = new AbortControllerShim();
+const signal = controller.signal;
+const signalShim = controllerShim.signal;
+
+signalCallback(signal);
+signalCallback(signalShim);
+
+signalShimCallback(signal);
+signalShimCallback(signalShim);
+
+controllerCallback(controller);
+controllerCallback(controllerShim);
+
+controllerShimCallback(controller);
```

Comment:
We can assign `AbortControllerShim` value to `AbortController` variable, but we cannot do the reverse.